storage/storage.go

```diff
@@ -0,0 +1,71 @@
+package storage
+
+import (
+	"strconv"
+	"sync"
+
+	"github.com/artemis13/platform-go-challenge/models"
+)
+
+var (
+	users = make(map[uint]*models.User)
+	mu    sync.RWMutex
+)
+
+// ConvertStringToUint safely converts a string to uint
+func ConvertStringToUint(s string) (uint, error) {
+	id, err := strconv.ParseUint(s, 10, 32)
+	if err != nil {
+		return 0, err
+	}
+	return uint(id), nil
+}
+
+// GetUser safely retrieves a user from the map
+func GetUser(userID uint) (*models.User, bool) {
```

Comment:
What would happen if the returned model was mutated by the caller outside this function? Is this a safe way to return this struct?

Reply:
I think I’ve made a big mistake here. The way that it is implemented the caller can affect the original data. Also race conditions are not being avoided here.  We should avoid returning a pointer to the user struct here and instead return a copy of the user.